rust/src/editor.rs

```diff
@@ -179,17 +187,24 @@ impl Editor {
     fn reset_contents(&mut self, new_contents: Rope) {
         self.engine = Engine::new(new_contents);
         self.text = self.engine.get_head();
-        self.dirty = true;
+        self.text_dirty = true;
         self.view.reset_breaks();
         self.set_cursor(0, true);
     }
 
     // render if needed, sending to ui
-    pub fn render(&mut self, tab_ctx: &TabCtx) {
-        if self.dirty {
+    pub fn render(&mut self) {
+        if self.text_dirty || self.view_dirty {
+            let tab_ctx = self.tab_ref.lock().unwrap();
             tab_ctx.update_tab(&self.view.render(&self.text, self.scroll_to));
-            self.dirty = false;
             self.scroll_to = None;
+            if self.text_dirty {
+                for plugin in &self.plugins {
+                    plugin.update();
```

Comment:
Do you plan to send some more information to the plugin later on? Like sending the same notification the frontend gets (the result of `&self.view.render(&self.text, self.scroll_to)`)?